### DISCUSSION TITLE
Text Element custom edit events

### DISCUSSION THREAD

**Discussion by zsviczian** [excalidraw/excalidraw#3776] [Q&A]
Hi! Can I capture the event when the user starts to edit a text element?
I want change the text right at the point in time when the edit starts, and change it again when the edit finishes.

My use-case is to implement native markdown links that render nicely... i.e. you click a text element and Obsidian opens another drawing or it opens a document. This link feature already works. 
What I want now is that while editing, the text reads "[[This is a link|alias]]" i.e. with full markdown markup, but when the user finishes editing the text is parsed and changed to "[[alias]]" or just "alias" or ":link:alias".

I can implement my own text element editor overlay, or I can implement ALT+click to toggle a text element between markup and rendered view, thus the user first ALT+clicks, then double clicks to edit the markup, then ALT+clicks again to toggle the text element to rendered view... but the nicest would be if I could capture the edit event and alter the text at that point in time.

Any suggestion or idea is most welcome!

**Comment by ad1992** [comment]
@zsviczian currently we don't support this. But from the requirement, it looks like you need `onBeforeTextEdit` and `onBeforeTextSubmit` callbacks which will be called before a user starts editing text and submitting text respectively.
That sounds like an interesting feature request!

**Reply by zsviczian** [reply]
That is exactly what I would need. Should I open an issue for this?

**Reply by ad1992** [reply]
@zsviczian sure you can create an issue from this discussion

**Reply by ad1992** [reply]
The progress can be tracked in https://github.com/excalidraw/excalidraw/issues/3779
